File: insertion-sort.js
let insertionSort = array => {
  for (let i = 1; i < array.length; i++) {
    let tempIndex = array[i];
    for (let j = i - 1; j >= 0 && array[j] > tempIndex; j--) {
      array[j + 1] = array[j];
    }
    array[j + 1] = tempIndex;
  }
  return array;
}


// reference:
// https://repl.it/@EvaB/JSInsertionSortcode

File: insertion_sort.py
def insertion_sort(list):
    for index in range(len(list)):
        temp_value = list[index]
        position = index
        while position > 0 and list[position - 1] > temp_value:
            list[position] = list[position - 1]
            position = position - 1
        list[position] = temp_value
    return list

# reference
# https://repl.it/@EvaB/InsertionSortPythonreview
